Tk 340: Pouvoir passer d'une plateforme à l'autre sans se déconnecter / reconnecter

2snufkin opened the pull request:
J'ai analysé la demande et identifié l'objectif.
J'ai vérifié la demande #[TK-526](https://tumorotek.myjetbrains.com/youtrack/issue/TK-526/Afficher-en-haut-des-ecrans-le-nom-de-la-plateforme-en-cours) pour comprendre les attentes.
J'ai ajouté un bouton entre le nom de la plateforme et la liste des collections.
J'ai configuré ce bouton pour rediriger l'utilisateur vers la page de sélection de plateforme et collections.
J'ai testé la fonctionnalité pour m'assurer du bon fonctionnement.

**pour eviter le hardcoding,  j'ai créé des variables au lieu de coder en dur.**

Dans ConfigManager, j'ai ajouté une URL.
Dans SessionUtils, j'ai défini les constantes correspondant à la clé de SessionScope.

Une tâche intéressante serait de créer une tâche bénéfique pour améliorer la qualité du code : progressivement supprimer (pas entièrement, car il y en a beaucoup) le hardcoding dans la base de code. Par exemple, tous les endroits où l'on accède à l'attribut de session en le codant en dur pourraient être remplacés par une attribution à une Enum. Dans ce cas, j'ai préféré utiliser des variables de classe.

Commit:
adding icon

2snufkin committed:
TK-340: Front

2snufkin committed:
TK-340: BACKEND and frontent correction

HUET-Sesan commented on tumorotek-core/src/main/java/fr/aphp/tumorotek/manager/ConfigManager.java:
```diff
@@ -79,6 +79,10 @@ public class ConfigManager
 
+   /// URLs de l'application
+   public static final String SELECT_BANQUE_URL = "/login/SelectBanque.zul";
+
```
comme vu ensemble, l'utilisation d'une constante est une bonne pratique.
Par contre, il faut l'utiliser dans les 3 classes où l'url est en dur. Cela permet d'être cohérent et de retrouver tous les cas d'appel avec une seule recherche. Il faut faire la modification dans cette PR avec un commit dédié

2snufkin replied:
V

HUET-Sesan commented on tumorotek-webapp/src/main/java/fr/aphp/tumorotek/action/MainWindow.java:
```diff
@@ -250,6 +252,10 @@ public void doAfterCompose(final Component comp) throws Exception{
          plateformeLabel.setValue(pf.getNom());
+
+         Image changePlateformeImage = (Image) mainBorderLayout.getFellow("northTopBanniere").getFellow("changePlateforme");
+         changePlateformeImage.setVisible(true);
```
L'auto-wiring de zk ne fonctionnant pas dans cette classe (ni Mathieu ne moi ne comprenons pourquoi), le fait de définir un attribut de classe du même nom que l'id du composant dans le zul (plateformeLabel) ne sert à rien. Or plateformeLabel n'est utilisé que dans cette méthode donc il vaut mieux définir plateformeLabel car une variable de la méthode. Ca permet d'être cohérent avec ce qui a été fait pour changePlateformeImage.
Par contre, il faut renommer l'id du zul changePlateforme en changePlateformeImage pour respecter la convention de nommage qui consiste à mettre le type du composant en suffixe.

HUET-Sesan commented on tumorotek-webapp/src/main/java/fr/aphp/tumorotek/action/MainWindow.java:
```diff
@@ -1430,6 +1439,16 @@ public void onSelectScanAndDelete(final Event ev){
+      } catch (Exception e) {
+         log.error("Erreur lors du changement de plateforme", e);
+      }
+   }
```
comme vu ensemble, il ne faut pas faire de try catch car cela cache l'erreur à l'utilisateur.
Par ailleurs, les 2 lignes de remove sont inutiles et peuvent donc être trompeuses car la session sera automatiquement mise à jour lors de la sélection de la nouvelle plateforme. Il faut donc les supprimer.
Cette méthode ne doit donc contenir que **Executions.sendRedirect(ConfigManager.SELECT_BANQUE_URL);**

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/main/main.zul:
```diff
@@ -254,6 +254,11 @@
 					<label id="plateformeLabel" sclass="formLabel" style="color: #001A61" />
+					<!-- Image permettant à l'utilisateur de changer de banque sans se déconnecter -->
+					<image id="changePlateforme" src="/images/icones/change_red.png"
+						   tooltiptext="${c:l('generale.changement.plateforme')}"
```
renommer changePlateforme en changePlateformeImage (voir commentaire dans la méthode MainWindow.doAfterCompose()

2snufkin committed:
TK-430: Using SELECT_BANQUE_URL

2snufkin committed:
TK-430: refactoring variables names

2snufkin committed:
TK-430: adding JavaDoc

2snufkin committed:
TK-430: Removing unused Var

2snufkin committed:
TK-430: fixing space

HUET-Sesan committed:
TK-340 : suppression constantes

ces constantes ont été introduites par les premiers dev de TK-340 mais ne sont finalement plus nécessaires
